File: app/src/main/java/com/example/kakaogalleryproject/SplashActivity.java
package com.example.kakaogalleryproject;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

public class SplashActivity extends Activity {
    private static int SPLASH_TIME_OUT = 1000;

    /** Called when the activity is first created. */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        new Handler().postDelayed(new Runnable(){
            @Override
            public void run(){
                showMain();
            }
        }, SPLASH_TIME_OUT);
    }
    public void showMain(){
        Intent intent = new Intent(getBaseContext(), MainActivity.class);
        startActivity(intent);
        overridePendingTransition(0, 0);
        finish();
        overridePendingTransition(0, R.anim.fade_out);
    }
}

File: app/src/main/java/com/example/kakaogalleryproject/ui/splash/SplashActivity.kt
package com.example.kakaogalleryproject.ui.splash

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import com.example.kakaogalleryproject.R
import com.example.kakaogalleryproject.constants.SPLASH_TIME_OUT
import com.example.kakaogalleryproject.ui.main.MainActivity

class SplashActivity : Activity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Handler().postDelayed({
            finish() // move on to main activity
            overridePendingTransition(0, R.anim.fade_out)
        }, SPLASH_TIME_OUT.toLong())
    }
}